### PR TITLE
Check for undefined values in BlueBlueVerified

### LINKED ISSUE #84: Access User Members Safely
[`BlueBlueVerified`](https://github.com/kheina-com/Blue-Blocker/blob/main/shared.js#L235-L303) should be updated such that all references are made safely.

ie `user.legacy.verified` becomes `user?.legacy?.verified`

### PR DESCRIPTION
**Pull request by jaredkotoff**
Closes #84 

Tested by adding to chrome and watching it block accounts
Also used this CodePen that shows how it handles a lot of missing data on user object: https://codepen.io/jaredkotoff/pen/MWPXYMP

Suggestions: 
* adding packjson.json and using jest for testing
* Typescript would also help avoid a lot of issues like this
  * This could be helpful: https://github.com/guocaoyi/create-chrome-ext

### CODE REVIEW

File: shared.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ export function SetOptions(items) {
 	options = items;
 }
 
-function unblockUser(user, user_id, headers, reason) {
+function UnblockUser(user, user_id, headers, reason, attempt = 1) {
```
Comment (jaredkotoff) [inline shared.js:16]: - Renamed for consistency with the rest of the file. 
- `attempt` was being used but never defined, like in `BlockUser()`

File: shared.js [hunk 2]
```diff
@@ -233,11 +233,19 @@ function BlockUser(user, user_id, headers, reason, attempt=1) {
...
+		console.error(logstr, 'invalid user object passed to BlockBlueVerified');
+		return;
+	}
+	const legacyName = user.legacy?.name;
+	const screenName = user.legacy?.screenName;
+	const formattedUserName = `${legacyName} (@${screenName})`;
+
 	// since we can be fairly certain all user objects will be the same, break this into a separate function
-	if (user.legacy.verified_type && !blockableVerifiedTypes.has(user.legacy.verified_type)) {
+	if (!blockableVerifiedTypes.has(user.legacy?.verified_type)) {
```
Comment (jaredkotoff) [inline shared.js:245]: initial check was unneeded. `.has()` will handle undefined and return false

File: shared.js [hunk 3]
```diff
@@ -233,11 +233,19 @@ function BlockUser(user, user_id, headers, reason, attempt=1) {
 const blockableAffiliateLabels = new Set(["AutomatedLabel"]);
 const blockableVerifiedTypes = new Set(["Business"]);
 export function BlockBlueVerified(user, headers) {
+	if (!user?.rest_id) {
```
Comment (jaredkotoff) [inline shared.js:236]: This verifies that there is a `user` object and has the critical `rest_id`
Other checks don't need to check if `user` is defined after this

File: shared.js (hunk 1 above)
Comment (DanielleMiu) [reply shared.js:16]: the idea behind the naming convention is going more toward's golang's styling of having exported objects be PascalCase and non-exported objects being camelCase. obviously I don't follow that very closely, but that was the idea. good catch on the attempt missing!

File: shared.js [hunk 4]
```diff
@@ -286,18 +292,24 @@ export function BlockBlueVerified(user, headers) {
...
+			console.log(logstr, `did not block user with NFT avatar ${formattedUserName} because they follow you.`);
 		}
 		else {
 			QueueBlockUser(user, String(user.rest_id), headers, ReasonNftAvatar);
 		}
 	}
+
+	function formatLegacyName(user) {
+		const legacyName = user.legacy?.name;
+		const screenName = user.legacy?.screenName;
```
Comment (DanielleMiu) [inline shared.js:312]: I think this object is wrong, I think it's `screen_name` because it's json formatting

File: shared.js [hunk 5]
```diff
@@ -286,18 +292,24 @@ export function BlockBlueVerified(user, headers) {
...
 		) {
-			console.log(logstr, `did not block user with NFT avatar ${user.legacy.name} (@${user.legacy.screen_name}) because they follow you.`);
+			console.log(logstr, `did not block user with NFT avatar ${formattedUserName} because they follow you.`);
 		}
 		else {
 			QueueBlockUser(user, String(user.rest_id), headers, ReasonNftAvatar);
 		}
 	}
+
+	function formatLegacyName(user) {
```
Comment (DanielleMiu) [inline shared.js:310]: is this function defined inside of `BlockBlueUser`?

**Review by DanielleMiu — COMMENTED** [review]
this looks pretty good! I'm really interested in your testing details, if you could add them to the pr description

File: shared.js (hunk 1 above)
Comment (jaredkotoff) [reply shared.js:16]: I kind of thought that, but saw other non-exported functions. I think this was the only function in this file that was camelCase.

**Review by DanielleMiu — APPROVED** [review]

**Pull request merged**